src/QirRuntime/lib/Tracer/tracer.hpp

```diff
@@ -105,10 +113,19 @@ namespace Quantum
         // Adds operation with given id into the given layer. Assumes that duration contraints have been satisfied.
         void AddOperationToLayer(OpId id, LayerId layer);
 
-        // Update the qubit state with the new layer information
+        // Update the qubit state with the new layer information.
         void UpdateQubitState(Qubit q, LayerId layer, Duration opDuration);
 
+        // Considers global barriers and conditional fences to find the barrier currently in effect.
+        LayerId GetEffectiveBarrier() const;
+
+        // For the given results finds the latest layer of the measurements that produced the results.
+        LayerId FindLatestMeasurementLayer(long count, Result* results);
+
       public:
+        // INVALID LayerId is treated as -Infinity, and REQUESTNEW -- as +Infinity
+        static LayerId LaterLayer(LayerId l1, LayerId l2);

```

Comment:
Minor:
From the function name, function signature, and comment it is unclear what the function does with the params.
For a _public_ func I would prefer that to be clear without looking into the implementation.